images/macos/scripts/build/configure-autologin.sh

```diff
@@ -6,9 +6,20 @@
 ##              script was taken from https://github.com/timsutton/osx-vm-templates/blob/master/scripts/autologin.sh
 ################################################################################
 
-echo "Enabling automatic GUI login for the '$USERNAME' user.."
-python3 $HOME/bootstrap/kcpassword.py "$PASSWORD"
-/usr/bin/defaults write /Library/Preferences/com.apple.loginwindow autoLoginUser "$USERNAME"
+source ~/utils/utils.sh
+
+if is_Sonoma; then
+    echo "Enabling autologin for Sonoma"
+    sudo defaults delete /Library/Preferences/com.apple.loginwindow autoLoginUser
+    sudo rm /etc/kcpassword
+    sudo sysadminctl -autologin set -userName "$USERNAME" -password "$PASSWORD"
+    /usr/bin/defaults read /Library/Preferences/com.apple.loginwindow autoLoginUser
+    echo "Validated autologin for Sonoma"
```

Comment:
There is `sysadminctl -autologin status` command. Should we use it instead to validate autologin?

Reply:
`sudo sysadminctl -autologin set -userName "$USERNAME" -password "$PASSWORD`  This is not working as expected. Still seeing the login window. Hence validating few items.

Reply:
The issue is resolved if we set this flag to false : `autoLoginUserScreenLocked`